src/main/java/org/medicmobile/webapp/mobile/NotificationForegroundHandler.java

```diff
@@ -0,0 +1,61 @@
+package org.medicmobile.webapp.mobile;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+public class NotificationForegroundHandler {
+	public static final String DEBUG_TAG = "NOTIFICATION_HANDLER";
+	private final Handler handler = new Handler(Looper.getMainLooper());
+	private static final int INTERVAL_MILLIS = 5 * 60 * 1000; //5mins interval
+	private static final int INITIAL_EXECUTION_DELAY = 2_000;
+
+	private final Runnable task;
+	private final WebView webView;
+	private boolean isPageLoaded = false;
```

Comment:
Minor but I think `isTaskPosted` would be a more descriptive name of what this value is actually indicating.